components/net/bluetooth_thread.rs

```diff
@@ -48,6 +52,48 @@ macro_rules! return_if_cached(
     );
 );
 
+macro_rules! check_blacklist {
+    ($blacklist:expr, $uuid:expr, $func:expr, $sender:expr) => (
+        match $blacklist.is_excluded($uuid) {
+            Ok(option) => match option {
+                true => {
+                    return drop($sender.send(Err(String::from(EXCLUDE_ERR))));
+                },
+                false => {},
+            },
+            Err(error) => {
+                return drop($sender.send(Err(String::from(error))));
+            },
+        }
+        if $func == 1 {
```

Comment:
or just pass the actual function+error and then the `if-else` part can be replaced